src/components/ui/TaskTable.tsx

```diff
@@ -0,0 +1,209 @@
+import { Accordion, AccordionItem, AccordionTrigger, AccordionContent } from "@radix-ui/react-accordion";
+import ChevronDownIcon from "../../assets/icons/chevron-down.svg?react";
+import { cn, formatDueDate, formatEstimatePoint, formatStatus, getDueDateColor, getTodayDate } from "../../libs/utils";
+import Tags from "./Tags";
+import GridHorizontalIcon from "../../assets/icons/grid-horizontal.svg?react";
+import { useDragAndDrop } from "@formkit/drag-and-drop/react";
+import { useState, useEffect } from "react";
+
+interface TaskTableProps {
+  status: Status;
+  tasks: Task[] | undefined;
+}
+
+const TaskTable = ({ status, tasks }: TaskTableProps) => {
+  const [taskList, setTaskList] = useState<Task[]>(tasks ?? []);
+
+  const [updateTaskMutation] = useUpdateTaskMutation({
+    update(cache, { data }) {
+      if (!data?.updateTask) return;
+
+      cache.modify({
+        fields: {
+          tasks(existingTasks = [], { readField }) {
+            return existingTasks.map((taskRef: any) => {
+              if (readField("id", taskRef) === data.updateTask.id) {
+                return { ...taskRef, ...data.updateTask };
+              }
+              return taskRef;
+            });
+          },
+        },
+      });
+    },
+  });
+
+  const handleDrop = async (draggedTask: Task, newStatus: Status) => {
+    try {
+      if (draggedTask.status === newStatus) {
+        console.log("Task moved within the same status");
+        return;
+      }
+      await updateTaskMutation({
+        variables: {
+          input: {
+            id: draggedTask.id,
+            status: newStatus,
+          },
+        },
+      });
+    } catch (error) {
+      console.error("Failed to update task status:", error);
+    }
+  };
+
+  const handleReorder = async (reorderedTasks: Task[]) => {
+    try {
+      const updatedTasks = reorderedTasks.map((task, index) => ({
+        ...task,
+        position: index,
+      }));
+
+      setTaskList(updatedTasks);
+
+      for (const task of updatedTasks) {
+        await updateTaskMutation({
+          variables: {
+            input: {
+              id: task.id,
+              position: task.position,
+            },
+          },
+        });
+      }
+    } catch (error) {
+      console.error("Failed to reorder tasks:", error);
+    }
+  };
+
+  const [taskListRef, _taskItems, setTaskItems] = useDragAndDrop<HTMLTableSectionElement, Task>(taskList, {
+    group: "taskList",
+    dragHandle: ".kanban-handle",
+    onDragend: async (selectedTask) => {
+      const draggedTask = selectedTask.draggedNode.data.value as Task;
+      const newStatus = selectedTask.parent?.el?.id as Status;
+
+      if (newStatus && draggedTask.status !== newStatus) {
+        await handleDrop(draggedTask, newStatus);
+      } else {
+        await handleReorder(selectedTask.values as Task[]);
+      }
+    },
+  });
+
+  useEffect(() => {
+    setTaskList(tasks ?? []);
```

Comment:
why an extra state is need? can't we use the data from the server to achieve this?, could you explain it please?